docs/sources/reference/api/docker_remote_api_v1.17.md

```diff
@@ -802,6 +804,138 @@ Status Codes:
+        the path specifies a directory and a volume is mounted lower in that
+        directory hierarchy, the volume's contents are not included in the
+        archive. Required.
+
+**Example request**:
+
+        GET /containers/8cce319429b2/copy?path=/foo/ HTTP/1.1
+
+**Example response**:
+
+        HTTP/1.1 200 OK
+        Content-Type: application/x-tar
+
+        {{ TAR STREAM }}
+
+Status Codes:
+
+-   **200** – no error
+-   **400** - client error, details in JSON response body, one of:
+        - path cannot be empty (missing path query parameter)
+        - no such file or directory (path does not exist)
+        - not a directory (path ends with a separator but exists as a file)
+-   **404** – no such container
+-   **500** – server error
+
+### Upload files or folders to a container from a Tar Archive.
+
+`PUT /containers/(id)/copy`
+
+Upload an archive of a resource to be extracted to a path in the filesystem
+of container `id`.
+
+Query Parameters:
+
+-   **srcPath** - Specifies the original path from which the payload archive
+        was created. Used as a hint to the server when determining if the
+        contents of the archive are a single file or should be interpreted
+        as directory contents. If the first entry of the archive has a name
+        matching the basename of **srcPath** then it is determined to be a
+        single file and copied to **dstPah** otherwise the archive is
+        interpreted as directory contents. Requried.
+-   **dstPath** - Specifies where in the container to extract the archive to.
+        If **dstPath** ends with a path separator and does not exist, the
+        directory is created. If **dstPath** is or is in a volume mount point
+        then the archive is extracted into the volume. If **dstPath**
+        specifies a directory and a volume is mounted lower in that directory
+        hierarchy then the archive will not be extracted into that volume.
+        Required.
+
+**Example request**:
+
+        PUT /containers/8cce319429b2/copy?srcPath=hello.txt&dstPath=/vol1 HTTP/1.1
+        Content-Type: application/x-tar
+
+        {{ TAR STREAM }}
+
+**Example response**:
+
+        HTTP/1.1 200 OK
+
+Status Codes:
+
+-   **200** – no error
+-   **400** - client error, details in JSON response body, one of:
+        - srcPath cannot be empty (missing srcPath query parameter)
+        - dstPath cannot be empty (missing dstPath query parameter)
+        - not a directory (dstPath ends with a separator but exists as a file)
+        - cannot copy directory (srcPath is a directory but dstPath exists and
+          is a file)
+-   **404** – no such container
+-   **500** – server error
+
+### Copy files or folders from one container to another.
+
+`PUT /containers/(id)/copy-accross`
+
+Copy resources from the filesystem of container `id` to a path in another
+container.
+
+Query Parameters:
+
+-   **dstContainer** - The container which is the recipient of the copied data.
```

Comment:
Am I correct in that we're doing PUT to the src container? Isn't that backwards?  This will make the HTTP purist's heads explode  :-)   Why isn't it PUT to the destination and the srcContainer is in a query parameter on that?